File: ulyp-ui/src/main/java/com/ulyp/ui/renderers/RenderedThrowable.java
package com.ulyp.ui.renderers;

import com.ulyp.core.printers.NullObjectRepresentation;
import com.ulyp.core.printers.ThrowableRepresentation;
import com.ulyp.ui.RenderSettings;
import com.ulyp.ui.util.ClassNameUtils;
import com.ulyp.ui.util.StyledText;
import javafx.scene.Node;

import java.util.ArrayList;
import java.util.List;

import static com.ulyp.ui.util.CssClass.CALL_TREE_IDENTITY_REPR;
import static com.ulyp.ui.util.CssClass.CALL_TREE_TYPE_NAME;

public class RenderedThrowable extends RenderedObject {

    protected RenderedThrowable(ThrowableRepresentation representation, RenderSettings renderSettings) {
        super(representation.getType());

        String className = renderSettings.showTypes() ? representation.getType().getName() : ClassNameUtils.toSimpleName(representation.getType().getName());

        List<Node> children = new ArrayList<>();

        children.add(StyledText.of(className, CALL_TREE_TYPE_NAME));
        children.add(StyledText.of("(", CALL_TREE_IDENTITY_REPR));
        if (!(representation.getMessage() instanceof NullObjectRepresentation)) {
            children.add(RenderedObject.of(representation.getMessage(), renderSettings));
        }
        children.add(StyledText.of(")", CALL_TREE_IDENTITY_REPR));

        super.getChildren().addAll(children);
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/renderers/RenderedThrowable.kt
package com.ulyp.ui.renderers

import com.ulyp.core.printers.NullObjectRecord
import com.ulyp.core.printers.ThrowableRecord
import com.ulyp.ui.RenderSettings
import com.ulyp.ui.util.ClassNameUtils.toSimpleName
import com.ulyp.ui.util.CssClass
import com.ulyp.ui.util.StyledText.of
import javafx.scene.Node

class RenderedThrowable(representation: ThrowableRecord, renderSettings: RenderSettings) : RenderedObject(representation.type) {

    init {
        val className =
            if (renderSettings.showTypes()) representation.type.name else toSimpleName(representation.type.name)
        val children: MutableList<Node> = ArrayList()
        children.add(of(className, CssClass.CALL_TREE_TYPE_NAME))
        children.add(of("(", CssClass.CALL_TREE_IDENTITY_REPR))
        if (representation.message !is NullObjectRecord) {
            children.add(of(representation.message, renderSettings))
        }
        children.add(of(")", CssClass.CALL_TREE_IDENTITY_REPR))
        super.getChildren().addAll(children)
    }
}